### ISSUE TITLE
Asymmetric error in x and y for point-source modeling

### ISSUE THREAD

**Issue by dsluse** [sibirrer/lenstronomy#146]
If I am not mistaken, it is not possible to have asymmetric error in x and y for the image positions of the point-source. In practice, Point-source positions uncertainties are rarely symmetric.  It could be usefull to implement "image_position_uncertainty_x" and "image_position_uncertainty_y" in the "kwargs_likelihood" ?

**Comment by sibirrer** [comment]
Indeed, currently the uncertainties are symmetric Gaussian's only. One way to generalize is to feed in 2d covariance matrices (not dependent on what x- and y- directions are). Then the uncertainties are expressed relative to the coordinate system (i.e. RA/DEC) rather than a pixel orientation. Would that make sense and be useful for you?

**Comment by dsluse** [comment]
Yes. I do think that something in that spirit is the best, and closest to observational errors.

**Comment by sibirrer** [comment]
Hi Dom, in the pre-release_dev branch I changed the input from 'image_position_uncertainties' to 'image_position_cov_error' and in the code documentation you see the meaning of it and in the tests how you get the same results as before. Let me know if this does it for you. You can also change things in your fork on the pre-release_dev branch and make a pull request into my repo's pre-release_dev branch if you have something else in mind. For the next release I will merge it then into the master. Currently this change is not compatible with the latest release and thus I do not change it in the master branch.

**Comment by dsluse** [comment]
I only did a test with a PSO optimization and it runs and gives a sensible result. It would be useful to give a list of covariance matrices for each image instead of a single one for all the images. I looked into the code, but I am not confident I can implement it quickly and efficiently.

**Comment by sibirrer** [comment]
I see. I am quite busy the next couple of days. This should not be too hard to implement. So I leave this issue open. Bug me in case it's not done when you need it.

**Comment by dsluse** [comment]
Sure ! No problem. I can work on the short term with the current implementation.
